src/ORM/Marshaller.php

```diff
@@ -295,7 +295,9 @@ protected function _belongsToMany(Association $assoc, array $data, $options = []
             if (array_intersect_key($primaryKey, $row) === $primaryKey) {
                 $keys = array_intersect_key($row, $primaryKey);
                 if (count($keys) === $primaryCount) {
-                    $conditions[] = $keys;
+                    foreach ($keys as $key => $value) {
+                        $conditions[][$assoc->alias() . '.' . $key] = $value;
```

Comment:
You can do

```php
$target = $assoc->target();
foreach ($keys as $key => $value) {
  $conditions[$target->aliasfield($key)] = $value
}
```

It saves a few `__call()` methods, and uses a method that people can customize if necessary.